src/fides/config/config_proxy.py

```diff
@@ -85,3 +94,27 @@ def __init__(self, db: Session) -> None:
         self.notifications = NotificationSettingsProxy(db)
         self.execution = ExecutionSettingsProxy(db)
         self.storage = StorageSettingsProxy(db)
+        self.security = SecuritySettingsProxy(db)
+
+    def load_current_cors_domains_into_middleware(self, app: FastAPI) -> None:
+        """
+        Util function that loads the current CORS domains from
+        `ConfigProxy` into the  `CORSMiddleware` at runtime.
+        """
+        for mw in app.user_middleware:
+            if mw.cls is CORSMiddleware:
+                current_middleware_domains = (
+                    mw.options["allow_origins"]
+                    if mw.options["allow_origins"] is not None
+                    else []
+                )
+                current_config_proxy_domains = (
+                    self.security.cors_origins
+                    if self.security.cors_origins is not None
+                    else []
+                )
+                patch_domains = set(
+                    [*current_middleware_domains, *current_config_proxy_domains]
+                )
+
+                mw.options["allow_origins"] = [str(domain) for domain in patch_domains]
```

Comment:
i know i sorta pointed you toward this implementation on your previous iteration, but i hadn't thought of a consequence of this additive logic that now seems like it _could_ be problematic, unless i'm thinking about this wrong. 

with the way this is written, won't we never be able to _remove_ a domain that's been added at some point via config setting?  initially a domain will be added to the `allow_origins` list/set, and then wouldn't it essentially stay there forever, since it'd continue to be retrieved as a `current_middleware_domains` on all subsequent updates (until the service is restarted).

would there be a significant downside to switching the logic here to _replace_ the existing `allow_origins` list, so that it's exactly what's been specified in the `ConfigProxy`? i'm not sure if there's an initial set of `allow_origins` that we need to keep in place from the get-go...if so, i think this problem could be a bit tricky to solve. if not, then i feel like we'd be better off just _replacing_ the `cors_origins` with whatever's been specified in the `ConfigProxy`. 

i guess ultimately i'm wondering on the motivation for making this additive rather than just a replace. it seems to me like the additive logic is a bit at-odds with the way we set our config properties generally, which assumes a replace sort of functionality. i get worried that trying to reconcile those two can be a bit tricky.

sorry i hadn't thought of this on the previous iteration! but just wanna make sure we get this right 👍 

```suggestion
            if mw.cls is CORSMiddleware:
                current_config_proxy_domains = (
                    self.security.cors_origins
                    if self.security.cors_origins is not None
                    else []
                )
                
                mw.options["allow_origins"] = [str(domain) for domain in current_config_proxy_domains]
```
^ may not be exactly how you'd wanna code it, but just to illustrate what i think would lead to a bit more consistent and transparent behavior, if we can get away with it.